app/services/file_locator.rb

```diff
@@ -30,6 +30,14 @@ def initialize(uri)
     def object
       @object ||= Aws::S3::Object.new(bucket_name: bucket, key: key)
     end
+
+    def localize(dir = "/tmp")
+      new_dir = File.join dir, SecureRandom.uuid
+      new_path = File.join new_dir, File.basename(key)
+      FileUtils.mkdir_p new_dir
+      object.download_file new_path
```

Comment:
I'm curious.  Is this a blocking operation or does it download in the background?  If it is blocking, should we add a comment saying that it could be a long running method?

Reply:
There should also be a comment about cleaning up `/tmp` in case it uses up available space.

Reply:
It's blocking. I can add a comment.